src/azure-cli-core/azure/cli/core/extension/operations.py

```diff
@@ -234,6 +235,7 @@ def add_extension(cmd, source=None, extension_name=None, index_url=None, yes=Non
                            "Please use with discretion.", extension_name)
         elif extension_name and ext.preview:
             logger.warning("The installed extension '%s' is in preview.", extension_name)
+        invalidate_command_index()
```

Comment:
@haroldrandom Please help verify this is the correct location to invalidate command index after changing extensions.